core/src/main/java/io/questdb/cairo/sql/RecordCursorFactory.java

```diff
@@ -139,8 +144,14 @@ default PageFrameCursor getPageFrameCursor(SqlExecutionContext executionContext,
         return null;
     }
 
-    default boolean hasDescendingOrder() {
-        return false;
+    /**
+     * Returns the direction of scanning used in this factory:
+     * - {@link #SCAN_DIRECTION_FORWARD}, {@link #SCAN_DIRECTION_BACKWARD} - for regular data/interval frame scans
+     * - {@link #SCAN_DIRECTION_OTHER} - for some index scans, e.g. cursor-order index lookup with multiple values
+     * where order is 'random'.
+     */
+    default int getScanDirection() {
```

Comment:
nit
```suggestion
    default int getTimestampScanDirection() {
```

Reply:
Please see the answer above .

Reply:
A lot of code readability below will benefit if this method had the word Timestamp in it, but I leave it with you to decide